### PR TITLE
xdstp: rename UDPA to xDS.

### PR DESCRIPTION
**Pull request by htuch**
As per the decision to move the cncf/udpa repository to cncf/xds branding.

Also updated cncf/udpa hash and updated identifier handling (moved from repeated to a flat string).

Risk level: Low (the only breaking API changes affect not-implemented-hide fields).
Testing: New unit tests for path components.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Review by mattklein123 — APPROVED** [review]
Awesome! This feels very much like the right move!

File: api/bazel/repository_locations.bzl [hunk 1]
```diff
@@ -22,14 +22,15 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
     ),
     com_github_cncf_udpa = dict(
-        project_name = "Universal Data Plane API",
-        project_desc = "Universal Data Plane API Working Group (UDPA-WG)",
+        project_name = "xDS API",
+        project_desc = "xDS API Working Group (xDS-WG)",
         project_url = "https://github.com/cncf/udpa",
-        version = "0.0.1",
-        sha256 = "83a7dcc316d741031f34c0409021432b74a39c4811845a177133f02f948fe2d8",
+        # Durin the UDPA -> xDS migration, we aren't working with releases.
```
Comment (moderation) [inline api/bazel/repository_locations.bzl:28]: s/Durin/During

**Review by moderation — APPROVED** [review]
Spelling change in the `repository_locations.bzl`. Deps LGTM

**Review by mattklein123 — APPROVED** [review]

**Comment by FeiYing9** [comment]
Can I ask that What’s the relationship between this [commit of 8c4a3c77](https://github.com/envoyproxy/envoy/commit/8c4a3c77a7de016a118aacc4cea933951b85e589) and commits in [issue#14157](https://github.com/envoyproxy/envoy/pull/14157/commits) ?

So which commits should I cherry-pick in my own forked project?

Looking forward to your reply, thanks.

**Pull request merged**